Ginger/Ginger/ALM/Repository/OctaneRepository.cs

```diff
@@ -75,27 +242,132 @@ public override List<string> GetTestPlanExplorer(string path)
+
+                        if (WorkSpace.Instance.Solution.MainApplication != null)
+                        {
+                            //add the applications mapped to the Activities
+                            foreach (Activity activ in tsBusFlow.Activities)
+                            {
+                                if (string.IsNullOrEmpty(activ.TargetApplication))
+                                {
+                                    if (tsBusFlow.TargetApplications.Where(x => x.Name == activ.TargetApplication).FirstOrDefault() == null)
+                                    {
+                                        ApplicationPlatform appAgent = WorkSpace.Instance.Solution.ApplicationPlatforms.Where(x => x.AppName == activ.TargetApplication).FirstOrDefault();
+                                        if (appAgent != null)
+                                        {
+                                            tsBusFlow.TargetApplications.Add(new TargetApplication() { AppName = appAgent.AppName });
+                                        }
+                                    }
+                                }
+                            }
+                            //handle non mapped Activities
+                            if (tsBusFlow.TargetApplications.Count == 0)
+                            {
+                                tsBusFlow.TargetApplications.Add(new TargetApplication() { AppName = WorkSpace.Instance.Solution.MainApplication });
+                            }
+                            foreach (Activity activ in tsBusFlow.Activities)
+                            {
+                                if (string.IsNullOrEmpty(activ.TargetApplication))
+                                {
+                                    activ.TargetApplication = tsBusFlow.MainApplication;
+                                }
+                            }
+                        }
+                        else
+                        {
+                            foreach (Activity activ in tsBusFlow.Activities)
+                            {
+                                activ.TargetApplication = null; // no app configured on solution level
+                            }
+                        }
+
+                        WorkSpace.Instance.SolutionRepository.AddRepositoryItem(tsBusFlow);
+                        Reporter.HideStatusMessage();
+                    }
+                    catch (Exception ex)
+                    {
+                        Reporter.ToUser(eUserMsgKey.ErrorInTestsetImport, testSetItemtoImport.TestSetName, ex.Message);
+                        Reporter.ToLog(eLogLevel.ERROR, "Error importing from QC", ex);
+                    }
+                }
+
+                Reporter.ToUser(eUserMsgKey.TestSetsImportedSuccessfully);
+
+                Reporter.ToLog(eLogLevel.DEBUG, "Imported from QC successfully");
```

Comment:
Even if there is exception in import we will show success message. This code should be inside the try at the end